File: base/pula-pula/.cache/lang/ts/trampoline.ts
import { Kid } from "./kid";

class Trampoline {
    private waiting: Kid[] = [];
    private playing: Kid[] = [];

    Trampoline() {
    }

    private removeFromList(name: string, list: Kid[]): Kid | null {
        return null;
    }

    arrive(kid: Kid): void {
    }

    enter(): void {
    }

    leave(): void {
    }

    removeKid(name: string): Kid | null {
        return null;
    }

    toString(): string {
        return  "[" + this.waiting.map(kid => kid.toString()).join(", ") + "]" + " => " + 
                "[" + this.playing.map(kid => kid.toString()).join(", ") + "]";
    }
}

export { Trampoline };

File: base/pula-pula/.cache/lang/java/Trampoline.java
import java.util.LinkedList;
import java.util.stream.Collectors;

public class Trampoline{
    private LinkedList<Kid> waiting;
    private LinkedList<Kid> playing;
    
    public Trampoline() {
        waiting = new LinkedList<>();
        playing = new LinkedList<>();
    }

    private Kid removeFromList(String name, LinkedList<Kid> list) {
        return null;
    }

    public void arrive(Kid kid) {
    }

    public void enter() {
    }

    public void leave() {
    }

    public Kid remoteKid(String name) {
        return null;
    }
    public String toString() {
        return   "[" + waiting.stream().map(Kid::toString).collect(Collectors.joining(", ")) + "]" + " => "
               + "[" + playing.stream().map(Kid::toString).collect(Collectors.joining(", ")) + "]";
    }
}
